tests/service/privacy_request/request_runner_service_test.py

```diff
@@ -412,6 +415,156 @@ def test_create_and_process_erasure_request_snowflake(
+    results = pr.get_results()
+    customer_table_key = (
+        f"EN_{pr.id}__access_request__redshift_example_test_dataset:customer"
+    )
+    assert len(results[customer_table_key]) == 1
+    assert results[customer_table_key][0]["email"] == customer_email
+    assert results[customer_table_key][0]["name"] == customer_name
+
+    address_table_key = (
+        f"EN_{pr.id}__access_request__redshift_example_test_dataset:address"
+    )
+
+    city = redshift_resources["city"]
+    state = redshift_resources["state"]
+    assert len(results[address_table_key]) == 1
+    assert results[address_table_key][0]["city"] == city
+    assert results[address_table_key][0]["state"] == state
+
+    pr.delete(db=db)
+
+
+@pytest.mark.integration_external
+def test_create_and_process_erasure_request_redshift(
+    redshift_example_test_dataset_config,
+    redshift_resources,
+    integration_config: Dict[str, str],
+    db,
+    cache,
+    erasure_policy,
+):
+    customer_email = redshift_resources["email"]
+    data = {
+        "requested_at": "2021-08-30T16:09:37.359Z",
+        "policy_key": erasure_policy.key,
+        "identity": {"email": customer_email},
+    }
+
+    # Should erase customer name
+    pr = get_privacy_request_results(db, erasure_policy, cache, data)
+    pr.delete(db=db)
+
+    connector = redshift_resources["connector"]
+    redshift_client = redshift_resources["client"]
+    with redshift_client.connect() as connection:
+        connector.set_schema(connection)
+        stmt = f"select name from customer where email = '{customer_email}';"
+        res = connection.execute(stmt).all()
+        for row in res:
+            # Confirms customer name has been replaced with null
+            assert row[0] is None
+
+        address_id = redshift_resources["address_id"]
+        stmt = f"select 'id', city, state from address where id = {address_id};"
+        res = connection.execute(stmt).all()
+        for row in res:
+            # Not yet masked because these fields aren't targeted by erasure policy
+            assert row[1] == redshift_resources["city"]
+            assert row[2] == redshift_resources["state"]
+
+    target = erasure_policy.rules[0].targets[0]
+    target.data_category = "user.provided.identifiable.contact.state"
+    target.save(db=db)
+
+    # Should erase state fields on address table
+    pr = get_privacy_request_results(db, erasure_policy, cache, data)
+    pr.delete(db=db)
+
+    connector = redshift_resources["connector"]
+    redshift_client = redshift_resources["client"]
+    with redshift_client.connect() as connection:
+        connector.set_schema(connection)
+
+        address_id = redshift_resources["address_id"]
+        stmt = f"select 'id', city, state from address where id = {address_id};"
+        res = connection.execute(stmt).all()
+        for row in res:
+            # Confirms address id/street are still there, but state fields have been replaced by null values.
```

Comment:
Great comment. A good extension could be to write a util method for these queries that could take the query results and unpack them into a dictionary to make it easier to read

Reply:
it looks like we can access ResultProxy results like they are dictionary keys, instead of accessing them as tuple elements, I'll refactor!